### PR TITLE
`with_priority` matcher for ActiveJob

### PR DESCRIPTION
**Pull request by mbajur**
Fixes https://github.com/rspec/rspec-rails/discussions/2757

### CODE REVIEW

File: spec/rspec/rails/matchers/active_job_spec.rb [hunk 1]
```diff
@@ -262,6 +262,18 @@ def perform; raise StandardError; end
...
+        hello_job.set(priority: 2).perform_later
+      }.to have_enqueued_job.with_priority(2)
+    end
+
+    it "passes with provided priority number as string" do
+      expect {
+        hello_job.set(priority: 2).perform_later
+      }.to have_enqueued_job.with_priority("2")
+    end
+
```
Comment (pirj) [inline spec/rspec/rails/matchers/active_job_spec.rb:276]: Can you please add a fee more cases? I can think of a failure case when we expect priority and it wasn’t set. Also when it was set, but doesn’t match the expectation. In both examples with checks if our failure message is reasonable.
I had another scenario on the tip of my tongue, but it keeps slipping. Can you think of other cases we should cover? You can use other similar qualifiers as a source of inspiration .

By the way, does this apply to mailers?

**Comment by pirj** [comment]
It also makes sense to add/extend scenario here https://github.com/rspec/rspec-rails/blob/4ab08fa954fe4e0ae6f2302fa7201dcad3f11521/features/job_specs/job_spec.feature#L63 
Those changes will appear here https://rspec.info/features/6-1/rspec-rails/job-specs/job-spec/, and some prefer those docs to the rdoc ones.

File: lib/rspec/rails/matchers/active_job.rb [hunk 2]
```diff
@@ -30,6 +31,11 @@ def on_queue(queue)
             self
           end
 
+          def with_priority(priority)
```
Comment (JonRowe) [inline lib/rspec/rails/matchers/active_job.rb:34]: Personally I would call this `priority` or `at_priority` to disambigute from `with`

**Comment by cmrd-senya** [comment]
ping @mbajur

**Comment by pirj** [comment]
Thank you @mbajur !

**Comment by JonRowe** [comment]
Released in 7.0.0

**Pull request merged**